dcgm/tests/conftest.py

```diff
@@ -0,0 +1,55 @@
+# (C) Datadog, Inc. 2023-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+import os
+from unittest import mock
+
+import pytest
+
+from datadog_checks.dcgm import DcgmCheck
+from datadog_checks.dev import docker_run
+from datadog_checks.dev.conditions import CheckDockerLogs, CheckEndpoints
+
+from . import common
+
+
+## Fixtures that will be included in a conftest.py folder:
+## TODO - Need to add the docker IP rather than hardcode the localhost
+@pytest.fixture(scope='session')
+def dd_environment():
+    compose_file = common.COMPOSE_FILE
+    conditions = [
+        CheckDockerLogs(identifier='caddy', patterns=['server running']),
+        CheckEndpoints(common.INSTANCE["openmetrics_endpoint"]),
+    ]
+    with docker_run(compose_file, conditions=conditions):
+        yield {
+            'instances': [common.INSTANCE],
+        }
+
+
+# For E2E and Unit testing:
+@pytest.fixture
+def instance():
+    return common.INSTANCE
```

Comment:
To stay on the safe side, you probably should deep copy this in case one test modifies the instance, which is global

```suggestion
    return copy.deepcopy(common.INSTANCE)
```

+ imports

Reply:
✅